skbio/stats/composition.py

```diff
@@ -1146,81 +1147,57 @@ def tree_basis(tree):
     return basis, nodes
 
 
-def _holm_bonferroni(p):
-    """Perform Holm-Bonferroni correction.
-
-    Perform Holm-Bonferroni correction ([1]_) for *p*-values to account for
-    multiple comparisons.
+def _calc_p_adjust(name, p):
+    """
+    Calculate the p-value adjustment for a given method.
 
     Parameters
-    ----------
-    p : ndarray of shape (n_tests,)
-        Original *p*-values.
+    -------
+        name (str): The name of the p-value adjustment method. Defaults to None.
+        p : ndarray of shape (n_tests,)
+            Original *p*-values.
 
     Returns
     -------
-    ndarray of shape (n_tests,)
-        Corrected *p*-values.
-
-    References
-    ----------
-    .. [1] https://en.wikipedia.org/wiki/Holm%E2%80%93Bonferroni_method
-
-    """
-    K = len(p)
-    sort_index = -np.ones(K, dtype=np.int64)
-    sorted_p = np.sort(p)
-    sorted_p_adj = sorted_p * (K - np.arange(K))
-    for j in range(K):
-        idx = (p == sorted_p[j]) & (sort_index < 0)
-        num_ties = len(sort_index[idx])
-        sort_index[idx] = np.arange(j, (j + num_ties), dtype=np.int64)
-
-    sorted_holm_p = [min([max(sorted_p_adj[:k]), 1]) for k in range(1, K + 1)]
-    holm_p = [sorted_holm_p[sort_index[k]] for k in range(K)]
-    return holm_p
-
-
-def _benjamini_hochberg(p):
-    """Perform Benjamini-Hochberg correction.
-
-    Perform Benjamini-Hochberg correction ([1]_) for *p*-values to account for
-    multiple comparisons.
+        list: The adjusted p-values.
 
-    Parameters
-    ----------
-    p : ndarray of shape (n_tests,)
-        Original *p*-values.
-
-    Returns
+    Raises
     -------
-    ndarray of shape (n_tests,)
-        Corrected *p*-values.
+        ValueError: If the given method name is not available.
+
+    See Also
+    --------
+    statsmodels.stats.multitest.multipletests
 
     References
     ----------
-    .. [1] https://en.wikipedia.org/wiki/Benjamini%E2%80%93Hochberg_procedure
+    .. [1] https://www.statsmodels.org/dev/generated/statsmodels.stats.multitest.multipletests.html
 
     """
-    # Derived from @Vladimir's answer at:
-    # https://stackoverflow.com/questions/7450957/
-    p = np.asarray(p)
-    K = len(p)
-    order = p.argsort()[::-1]
-    bh_p = p[order] * K / np.arange(K, 0, -1)
-    bh_p = np.minimum(1, np.minimum.accumulate(bh_p))
-    return bh_p[order.argsort()]
-
-
-def _dispatch_p_adjust(name):
-    """Refer to a *p*-value correction function by the method's name."""
     if name is None:
-        return
+        return p
     name_ = name.lower()
+
+    # Original options are kept for backwards compatibility
     if name_ in ("holm", "holm-bonferroni"):
-        return _holm_bonferroni
-    elif name_ in ("bh", "fdr_bh", "benjamini-hochberg"):
-        return _benjamini_hochberg
+        name_ = "holm"
+    if name_ in ("bh", "fdr_bh", "benjamini-hochberg"):
+        name_ = "fdr_bh"
+
+    if name_ in (
```

Comment:
Here, you explicitly listed all FDR correction methods available in statsmodels. This is great. However, it might be broken by future updates of statsmodels, if they decide to add/remove/change methods. How about the following:

```python
try:
    res = sm_multipletests(pvals=arr, alpha=0.05, method='xxx')
except ValueError as e:
    if 'method not recognized' in str(e)
        raise ValueError(f'{name} is not an available FDR correction method.')
    else:
        raise ValueError(f'Cannot perform FDR correction using the {name} method.')
else:
    return res[1]
```

Reply:
I have inserted the code. Thanks for pointing that out!